src/webui/api/torrentscontroller.cpp

```diff
@@ -801,8 +801,14 @@ void TorrentsController::setCategoryAction()
 
     const QStringList hashes {params()["hashes"].split('|')};
     const QString category {params()["category"].trimmed()};
-    applyToTorrents(hashes, [category](BitTorrent::TorrentHandle *torrent)
+    const QString savePath {params()["savePath"].trimmed()};
```

Comment:
setCategory is (should) used to assign torrent to existing category. Why this inconsistent parameter?

Reply:
This `setCategory` API receives requests for both new and existing categories. The `savePath` is used when the category is new.

Reply:
>The savePath is used when the category is new.

It's incorrect interface.

Reply:
It’s the existing interface. We can address it in a separate PR.

Reply:
Really? Either you don't know what the "interface" is, or you're just kidding... There are no incorrect parameters before your changes. `torrent.setCategory(category)` is correct interface but `torrent.setCategory(category, savePath)` isn't since "savePath" has nothing to do with "assign torrent(s) to category".